### PR TITLE
Add paging option to search result option

### PR DESCRIPTION
**Pull request by nuxzero**

### CODE REVIEW

File: src/main/java/co/omise/models/SearchResult.java [hunk 1]
```diff
@@ -73,6 +73,8 @@ public void setTotalPages(int totalPages) {
         private String query;
         private Map<String, String> filters;
         private Ordering order;
+        private int page;
+        private int perPage;
```
Comment (Namiii) [inline src/main/java/co/omise/models/SearchResult.java]: I know it's an API limitation, but just looks odd doing it this way. Don't think there's anything we can do though

File: src/main/java/co/omise/models/SearchResult.java [hunk 2]
```diff
@@ -124,10 +137,19 @@ public Options order(Ordering order) {
                     builder = builder.put("filters[" + entry.getKey() + "]", entry.getValue());
                 }
             }
+
             if (order != null) {
                 builder = builder.put("order", serializer.serializeToQueryParams(order));
             }
 
+            if (page > 0) {
+                builder = builder.put("page", page + "");
```
Comment (Namiii) [inline src/main/java/co/omise/models/SearchResult.java]: Could be better to use `String.valueOf(int)`

File: src/main/java/co/omise/models/SearchResult.java [hunk 3]
```diff
@@ -124,10 +137,19 @@ public Options order(Ordering order) {
...
+            }
+
+            if (perPage > 0) {
+                builder = builder.put("per_page", perPage + "");
```
Comment (Namiii) [inline src/main/java/co/omise/models/SearchResult.java]: Same as above

File: src/main/java/co/omise/models/SearchResult.java (hunk 2 above)
Comment (nuxzero) [reply src/main/java/co/omise/models/SearchResult.java]: 👌

File: src/main/java/co/omise/models/SearchResult.java (hunk 1 above)
Comment (pitiphong-p) [reply src/main/java/co/omise/models/SearchResult.java]: `perPage` name is not descriptive name for me. `numberOfItemsPerPage` would be ideal for me but not sure if it's too long

File: src/main/java/co/omise/models/SearchResult.java [hunk 4]
```diff
@@ -104,6 +106,16 @@ public Options order(Ordering order) {
             return this;
         }
 
+        public Options page(int page) {
+            this.page = page;
+            return this;
+        }
+
+        public Options perPage(int perPage) {
```
Comment (pitiphong-p) [inline src/main/java/co/omise/models/SearchResult.java]: `perPage` name here too

File: src/main/java/co/omise/models/SearchResult.java (hunk 1 above)
Comment (nuxzero) [reply src/main/java/co/omise/models/SearchResult.java]: @pitiphong-p This just perspective from Omise API. BTW `numberOfItemsPerPage` sound a bit  too long. What about `itemsPerPage`?

Comment (pitiphong-p) [reply src/main/java/co/omise/models/SearchResult.java]: Yeah, maybe. What's the common convention for a Java library?

Comment (Namiii) [reply src/main/java/co/omise/models/SearchResult.java]: That's also true

**Comment by nuxzero** [comment]
@pitiphong-p @Namiii So, can I close this and keep `perPage` as the same name?

**Comment by pitiphong-p** [comment]
@nuxzero Well, let's research on the convention a little bit or else we can go with `numberOfDataPerPage`?

**Comment by pitiphong-p** [comment]
Just did a quick look up. `dataPerPage` would be ok for this. Also there’s another name came up which is `pageSize`

**Comment by pitiphong-p** [comment]
Well then I think we would go with `dataPerPage`? @Namiii

**Comment by Namiii** [comment]
@nuxzero @pitiphong-p `dataPerPage` seems alright 👍

**Comment by nuxzero** [comment]
@pitiphong-p @Namiii Changed the `perPage` to `dataPerPage`. I'll solve conflicts then close this PR. 🙏

**Pull request merged**